plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1292,6 +1302,85 @@ public void dropColumn(ConnectorSession session, ConnectorTableHandle tableHandl
         metastore.dropColumn(hiveTableHandle.getSchemaName(), hiveTableHandle.getTableName(), columnHandle.getName());
     }
 
+    @Override
+    public void setColumnType(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle, Type type)
+    {
+        HiveTableHandle table = (HiveTableHandle) tableHandle;
+        failIfAvroSchemaIsSet(table);
+        HiveColumnHandle column = (HiveColumnHandle) columnHandle;
+
+        if (table.getPartitionNames().isPresent() && table.getPartitionNames().get().contains(column.getName())) {
+            throw new TrinoException(NOT_SUPPORTED, "Changing partition column types is not supported");
+        }
+        if (!canChangeColumnType(column.getType(), type)) {
+            throw new TrinoException(NOT_SUPPORTED, "Cannot change type from %s to %s".formatted(column.getType(), type));
+        }
+        metastore.setColumnType(table.getSchemaName(), table.getTableName(), column.getName(), toHiveType(type));
+    }
+
+    private static boolean canChangeColumnType(Type sourceType, Type newType)
+    {
+        if (sourceType.equals(newType)) {
+            return true;
+        }
+        if (sourceType == TINYINT) {
+            return newType == SMALLINT || newType == INTEGER || newType == BIGINT;
+        }
+        if (sourceType == SMALLINT) {
+            return newType == INTEGER || newType == BIGINT;
+        }
+        if (sourceType == INTEGER) {
+            return newType == BIGINT;
+        }
+        if (sourceType == REAL) {
+            return newType == DOUBLE;
+        }
+        if (sourceType instanceof VarcharType || sourceType instanceof CharType) {
+            return newType instanceof VarcharType || newType instanceof CharType;
```

Comment:
What about truncation? will we truncate on read?

what about varchar values with trailing spaces? after conversion to char, will we strip the spaces away?

Reply:
> What about truncation? will we truncate on read?

Could you share the test case example? 

> what about varchar values with trailing spaces? after conversion to char, will we strip the spaces away?

Trailing spaces aren't trimmed on read. I will add a test case.